src/actions/actionBoundText.tsx

```diff
@@ -129,18 +138,105 @@ export const actionBindText = register({
       }),
     });
     redrawTextBoundingBox(textElement, container);
-    const updatedElements = elements.slice();
-    const textElementIndex = updatedElements.findIndex(
-      (ele) => ele.id === textElement.id,
-    );
-    updatedElements.splice(textElementIndex, 1);
-    const containerIndex = updatedElements.findIndex(
-      (ele) => ele.id === container.id,
+
+    return {
+      elements: adjustZIndexForContainer(elements, textElement, container.id),
+      appState: { ...appState, selectedElementIds: { [container.id]: true } },
+      commitToHistory: true,
+    };
+  },
+});
+
+const adjustZIndexForContainer = (
+  elements: readonly ExcalidrawElement[],
+  textElement: ExcalidrawTextElement,
+  containerId: string,
+) => {
+  const updatedElements = elements.slice();
+  const textElementIndex = updatedElements.findIndex(
+    (ele) => ele.id === textElement.id,
+  );
+  updatedElements.splice(textElementIndex, 1);
+  const containerIndex = updatedElements.findIndex(
+    (ele) => ele.id === containerId,
+  );
+  updatedElements.splice(containerIndex + 1, 0, textElement);
+  return updatedElements;
+};
```

Comment:
in case of creating a container from text, it should put the container below the text index, not the other way round as with binding text to an existing container action.

Reply:
How does it make a difference as I am pushing the text element above the container? I am reusing the function for both cases

Reply:
we should respect the original z-index (of the text in this case):

![excal-wrap-into-container-z-index-issue](https://user-images.githubusercontent.com/5153846/222526191-924ad425-f5da-4fd5-aef2-bd3b0e81d7c6.gif)

Reply:
True, I have fixed it